test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -836,6 +837,13 @@ private static String randomJodaAndJavaSupportedTimezone(List<String> zoneIds) {
             () -> randomFrom(zoneIds));
     }
 
+    /**
+     * Generate a random valid date formatter pattern.
+     */
+    public static String randomDateFormatterPattern() {
+        return randomFrom(FormatNames.values()).getName();
```

Comment:
I think we could generate even more randomness here but this feels like enough for now.